### PR TITLE
Resizing based on ref and window listeners

### PR DESCRIPTION
**Pull request by beweinreich**
- Made both pages class based. 
- Wrapped them in a div with a ref (since you cant ref a functional component)
- Attached window listeners to change `onResize`
- Change width of sidebar

### CODE REVIEW

File: src/components/dashboard/index.tsx [hunk 1]
```diff
@@ -195,137 +195,167 @@ function DashboardMainScrollViewContent({
...
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.onResize.bind(this));
+  }
 
-  onDashboardChangeWeek,
-  onChangeSidebarVisibility,
-  onCloseModal,
-  onShowModal,
-}) {
+  onResize() {
```
Comment (1egoman) [inline src/components/dashboard/index.tsx]: If you make this function an attribute on the class with `onResize = () => {}` you don't need the `.bind(this)`s when you call it!

File: src/components/dashboard/index.tsx [hunk 2]
```diff
@@ -195,137 +195,167 @@ function DashboardMainScrollViewContent({
...
 }
 
-export function Dashboard({
-  dashboards,
-  selectedDashboard,
-  activeModal,
+export class Dashboard extends React.Component<any, any> {
+  private pageContainerRef: React.RefObject<HTMLInputElement>;
+
+  constructor(props) {
```
Comment (1egoman) [inline src/components/dashboard/index.tsx]: Kinda nitpicky and not all that important, but you can define these attributes inine and don't really need to define a constructor here. Doing it this way also means that you don't need to define the explicit type on `pageContainerRef`.

```
pageContainerRef = React.createRef();
state = {
  pageSize: 0,
};
```

Not all that important though, what you have works fine and is just as easy to understand!

File: src/components/explore/index.tsx [hunk 3]
```diff
@@ -147,93 +147,124 @@ function pruneHierarchy(spaceTree, matchedSpaceIds) {
...
-  let filteredSpaces = spaceHierarchy.data;
-  if (spaces.filters.search) {
-    const matchedSpaceIds = fuzzy.filter(
-      spaces.filters.search,
-      spaces.data,
-      { pre: '<', post: '>', extract: x => x['name'] }
-    ).map(x => x.original['id']);
-    const filteredSpacesCopy = JSON.parse(JSON.stringify(filteredSpaces));
-    filteredSpaces = filteredSpacesCopy.map(x => pruneHierarchy(x, matchedSpaceIds)).filter(x => x);
+export class Explore extends React.Component<any, any> {
```
Comment (1egoman) [inline src/components/explore/index.tsx]: Same notes on this component as the other one

File: src/components/dashboard/index.tsx (hunk 2 above)
Comment (beweinreich) [reply src/components/dashboard/index.tsx]: Makes sense

**Pull request merged**